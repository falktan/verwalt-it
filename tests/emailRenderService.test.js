import { jest } from '@jest/globals';
import 'dotenv/config';

// Mock the token utility to return deterministic values
jest.unstable_mockModule('../api/utils/token.js', () => ({
  createAccessToken: jest.fn(({ userRole }) => { return 'test-access-token'})
}));

// Import after mocking (required for ES6 modules)
const { renderEmailsNewSubmission } = await import('../api/emailRenderService.js');

describe('Email Render Service', () => {
  const mockFormData = {
    vorname: 'Max',
    nachname: 'Mustermann',
    email: 'max.mustermann@example.com',
    betreuer_betrieblich_email: 'betreuer.betrieblich@example.com',
    betreuer_hochschule_email: 'betreuer.hochschule@example.com',
    betreuer_korreferent_email: 'betreuer.korreferent@example.com'
  };

  const mockSecrets = {
    submissionId: 'test-submission-123',
    formEncryptionSecret: 'test-encryption-secret'
  };

  beforeEach(() => {
    // Set up environment variables
    process.env.BASE_URL = 'https://test.example.com';
    process.env.EMAIL_PRUEFUNGSAMT_TO = 'pruefungsamt@example.com';
    process.env.ACCESS_TOKEN_ENCRYPTION_SECRET = 'test-access-token-secret';
  });

  afterEach(() => {
    // Clean up environment variables
    delete process.env.BASE_URL;
    delete process.env.EMAIL_PRUEFUNGSAMT_TO;
    delete process.env.ACCESS_TOKEN_ENCRYPTION_SECRET;
  });

  test('should render all emails correctly', () => {
    const result = renderEmailsNewSubmission({ formData: mockFormData, secrets: mockSecrets });
    
    expect(result).toMatchSnapshot();
  });
});