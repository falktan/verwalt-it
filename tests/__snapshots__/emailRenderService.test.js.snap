// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Email Render Service should render all emails correctly 1`] = `
[
  {
    "body": "
Sehr geehrte Frau/Herr Max Mustermann,
Vielen Dank für die Anmeldung der Abschlussarbeit.
Über folgenden Link können Sie den Status der Anmeldung verfolgen:
https://test.example.com/hochschule-schmalkalden/fak-elektrotechnik/antrag-ausgabe-batchelorarbeit/formular?token=test-access-token
",
    "email_to": [
      "max.mustermann@example.com",
    ],
    "subject": "Anmeldung der Abschlussarbeit",
  },
  {
    "body": "
Liebes Prüfungsamt,
von Max Mustermann wurde soeben das Formular zur Anmeldung der Abschlussarbeit ausgefüllt.
Bitte bearbeiten Sie die Anfrage über folgenden Link:
https://test.example.com/hochschule-schmalkalden/fak-elektrotechnik/antrag-ausgabe-batchelorarbeit/formular?token=test-access-token
",
    "email_to": [
      "pruefungsamt@example.com",
    ],
    "subject": "Anmeldung der Abschlussarbeit",
  },
  {
    "body": "
Sehr geehrte Damen und Herren,
von Max Mustermann wurde soeben das Formular zur Anmeldung der Abschlussarbeit ausgefüllt.
Sie wurden als betrieblicher Betreuer angegeben.
Bitte bestätigen Sie über folgenden Link die Anmeldung der Abschlussarbeit:
https://test.example.com/hochschule-schmalkalden/fak-elektrotechnik/antrag-ausgabe-batchelorarbeit/formular?token=test-access-token
",
    "email_to": [
      "betreuer.betrieblich@example.com",
    ],
    "subject": "Anmeldung der Abschlussarbeit",
  },
  {
    "body": "
Sehr geehrte Damen und Herren,
von Max Mustermann wurde soeben das Formular zur Anmeldung der Abschlussarbeit ausgefüllt.
Sie wurden als Hochschulbetreuer angegeben.
Bitte bestätigen Sie über folgenden Link die Anmeldung der Abschlussarbeit:
https://test.example.com/hochschule-schmalkalden/fak-elektrotechnik/antrag-ausgabe-batchelorarbeit/formular?token=test-access-token
",
    "email_to": [
      "betreuer.hochschule@example.com",
    ],
    "subject": "Anmeldung der Abschlussarbeit",
  },
  {
    "body": "
Sehr geehrte Damen und Herren,
von Max Mustermann wurde soeben das Formular zur Anmeldung der Abschlussarbeit ausgefüllt.
Sie wurden als Korreferent angegeben.
Bitte bestätigen Sie über folgenden Link die Anmeldung der Abschlussarbeit:
https://test.example.com/hochschule-schmalkalden/fak-elektrotechnik/antrag-ausgabe-batchelorarbeit/formular?token=test-access-token
",
    "email_to": [
      "betreuer.korreferent@example.com",
    ],
    "subject": "Anmeldung der Abschlussarbeit",
  },
]
`;
